### PR TITLE
make: check if APPLICATION is set

### LINKED ISSUE #1282: make: building an application without setting APPLICATION succeeds
It results in a file named `.elf` .

### PR DESCRIPTION
**Pull request by OlegHahm**
Fixes #1282.

### CODE REVIEW

File: Makefile.include [hunk 1]
```diff
@@ -129,6 +129,9 @@ ifeq (,$(UNZIP_HERE))
   endif
 endif
 
+ifeq (, ${APPLICATION})
+    $(error An application name must be specified as APPLICATION.)
```
Comment (Kijewski) [inline Makefile.include]: Maybe:
```make
APPLICATION := $(shell basename "$(CURDIR)")
```

Comment (authmillenon) [inline Makefile.include]: :-1: there is already to much assumption going around that this is always the case... (just a feeling, not rational thought)

**Comment by authmillenon** [comment]
If nobody objects: ACK.

**Comment by OlegHahm** [comment]
@Kijewski, feel free to propose your solution in a different PR. :)

**Pull request merged**